test/async-hooks/test-async-local-storage-thenable.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const { AsyncLocalStorage } = require('async_hooks');
+
+// This test verifies that async local storage works with thenables
+
+const store = new AsyncLocalStorage();
+const data = Symbol('verifier');
+
+const then = common.mustCall((cb) => {
+  assert.strictEqual(store.getStore(), data);
+  setImmediate(cb);
+}, 4);
```

Comment:
Can you please explain why it needs to be called 4 times ?

Reply:
Because there are 4 testcases calling `store.run()` below which use a thenable in different variants.